### PR TITLE
Low hanging fruits of some auto fixable improvements via clang-tidy

### PR DESCRIPTION
**Pull request by AlexanderLanin**
**not ready to be merged**

What do you think? I could extend and clean this up if this goes to a reasonable direction.

- Incorporating clang-tidy into make/CI is out of scope here
- Do not bother with merge conflicts. This pull request can easily be regenerated.
- For easy review have a look at the individual commits
- All credits go to the clang-tidy team :-)
- This is not meant to solve everything. It just makes the code look a little more like C++

### CODE REVIEW

**Comment by llunak** [comment]
I like the the modernize-use-null and modernize-redundant-void-arg, I think they're simple and obvious. The .gitignore is trivial too.
I don't like the readability-simplify-boolean-expr one, although the code is technically the sema, semantically the code is different (there are many blocks of code that check many special cases and then return the "normal" return value at the end, so the code paths are not equal).
I like the general idea of modernize-redundant-void-arg, but I'm not a big fan of "auto", I think it makes code less readable by using a "magic" type. Can clang-tidy be told to spell out the type of the loop variable?

File: daemon/main.cpp [hunk 1]
```diff
@@ -591,9 +591,7 @@ bool Daemon::setup_listen_fds()
                 return false;
         }
     }
-    if( !setup_listen_unix_fd())
-        return false;
-    return true;
+    return setup_listen_unix_fd();
```
[the remark this answers is not in the document]
Comment (llunak) [reply daemon/main.cpp]: Yes.

**Comment by llunak** [comment]
> Sorry for the question, but as it costs time to do so... would anyone actually merge this if I were to manually improve this PR?

Yes.
Looking at the diff, code such as "for( size_t i = 0; ... cmake_checks[ i ] ..." already does not say what the type of the item is, so there is no regression when using auto, but changing code like "for( list<CompileServer *>::iterator it = " loses the type. So if it would be a lot of work to change all of them, please at least change those where the type already is spelled out now, that should be fairly simple to do just by looking at the diff.

Also please have a look at the Codacy automated review and the warnings it points out.

File: daemon/main.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexanderLanin) [reply daemon/main.cpp]: Looking at some additional lines than shown here by default I really feel this change is ok.
The function is called `setup_listen_fds`. It does some stuff. Then it returns `setup_listen_unix_fd`. Seems better this way. Now I see way better that it just returns the value of `setup_listen_unix_fd`.

**Comment by AlexanderLanin** [comment]
The Codacy warning are legitimate, however I have not touched the issues it warns about and I don't know in which direction to fix them.
The other jobs fail because of some wrong repository. This doesn't seem connect to this PR.

File: daemon/main.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (llunak) [reply daemon/main.cpp]: The setup_listen_fds() function calls 3x a setup_* function, and always does "return false" if any of the calls fail. So it is consistent there, and the purpose of the function is not to end up calling just setup_listed_unix_fd().

[the remark this answers is not in the document]
Comment (AlexanderLanin) [reply daemon/main.cpp]: oh now I see it

Comment (AlexanderLanin) [inline daemon/main.cpp]:
```suggestion
    if( !setup_listen_unix_fd())
        return false;
    return true;
```

**Comment by llunak** [comment]
You're right about the Codacy warnings, I'll fix them.

**Comment by llunak** [comment]
Are you still working on this, or do you consider this ready for submission? If the latter, can you please rewrite the history to drop the unwanted commits instead of adding another commit reverting them?

**Comment by AlexanderLanin** [comment]
I'm considering this ready for review.

However I disagree with the rewrite as the first commit is clang-tidy applied as is and the second is about changing (reverting) one very particular case of that rewrite. It's not a drop of the commit, only of one instance.
It makes sense to keep that separate, as merging them would indicate that clang-tidy did not change this one instance for no particular reason.

**Comment by AlexanderLanin** [comment]
Merging was not trivial, so I just reapplied clang-tidy with some cherry-picking.

**Comment by llunak** [comment]
Ok. Thank you.

**Pull request merged**